src/nimlsp.nim

```diff
@@ -118,8 +118,24 @@ proc getProjectFile(file: string): string =
       discard
     path = dir
 
+proc findProjectNimFile*( pkg: string): string =
+  var dir = pkg
+  let (_, pkgname, _) = splitFile(pkg)
+  for f in os.walkDirRec(dir, yieldFilter = {pcDir},relative = false ):
+    let x = changeFileExt(f / pkgname, ".nim")
+    if fileExists(x):
+      return x
+
+proc file2project(fileuri:string):string =
```

Comment:
The openFiles table is meant to map to project files, why is this necessary?

Reply:
I assuming `projectFiles` is project root-> nimsuggest instance mapping, so here have two concept project entry nim file,project root, now we have project entry nim file -> project root and project root  -> project entry nim file